src/prefect/engine/results/json_result.py

```diff
@@ -0,0 +1,57 @@
+import json
+
+from typing import Any
+
+from prefect.engine.result import Result
+
+
+class JSONResult(Result):
+    """
+    Hook for storing and retrieving JSON serializable Python objects that can
+    safely be stored directly in a Prefect database.
+
+    Args:
+        - value (Any): the underlying value this Result should represent
+    """
+
+    def __init__(self, value: Any = None, **kwargs: Any) -> None:
+        self.value = value
+        super().__init__(value=value, **kwargs)
+
+    def read(self, loc: str = None) -> Result:
+        """
+        Returns the underlying value regardless of the argument passed.
+
+        Args:
+            - loc (str): an unused argument
+        """
+        new = self.copy()
+        new.value = json.loads(loc or new.filepath)
```

Comment:
> in anticipation of exclusively serializing only Result.filepath to Cloud, so that Cloud can have a copy of the value to use for restart

Yes that's exactly correct.

Re: the naming idea, that's a great idea!  Let's definitely change the name.  Maybe `PrefectResult` to keep it short?

Reply:
:+1: Yes `PrefectResult` is short and sweet!